src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/PythonPreferencesPane.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * PythonPreferencesPane.java
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.prefs.views;
+
+import org.rstudio.core.client.Debug;
+import org.rstudio.core.client.ElementIds;
+import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.prefs.RestartRequirement;
+import org.rstudio.core.client.resources.ImageResource2x;
+import org.rstudio.core.client.widget.InfoBar;
+import org.rstudio.core.client.widget.OperationWithInput;
+import org.rstudio.core.client.widget.TextBoxWithButton;
+import org.rstudio.studio.client.server.ServerError;
+import org.rstudio.studio.client.server.ServerRequestCallback;
+import org.rstudio.studio.client.workbench.prefs.model.UserPrefs;
+import org.rstudio.studio.client.workbench.prefs.views.python.PythonInterpreterListEntryUi;
+import org.rstudio.studio.client.workbench.prefs.views.python.PythonInterpreterSelectionDialog;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.resources.client.ImageResource;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.inject.Inject;
+
+public class PythonPreferencesPane extends PreferencesPane
+{
+   @Inject
+   public PythonPreferencesPane(PythonDialogResources res,
+                                PythonServerOperations server)
+   {
+      res_ = res;
+      server_ = server;
+      
+      add(headerLabel("Python"));
+      
+      mismatchWarningBar_ = new InfoBar(InfoBar.WARNING);
+      mismatchWarningBar_.setText(
+            "The active Python interpreter has been changed by an R startup script.");
+      mismatchWarningBar_.setVisible(false);
+      add(spaced(mismatchWarningBar_));
+      
+      tbPythonInterpreter_ = new TextBoxWithButton(
+            "Python interpreter:",
+            "(No interpreter selected)",
```

Comment:
Should use constant PYTHON_PLACEHOLDER_TEXT to ensure they match?

Reply:
Yes, we should! Thanks for catching that.